### PR TITLE
Feature/494 oa woks all permissions

### PR DESCRIPTION
**Pull request by anusharanganathan**
This is a PR for https://github.com/antleaf/jct-project/issues/494

### CODE REVIEW

File: server/service/jct/api.coffee [hunk 1]
```diff
@@ -736,38 +736,141 @@ API.service.jct.permission = (issn, institution) ->
...
+      # evaluate each permission from OA works
+      oa_check =
+        OABCompliant: [],
+        OABIncomplete: [],
+        OABNonCompliant: []
+      for permission in perms.all_permissions
+        p_values = _gather_values(permission)
+        p_checks = _gather_checks(permission)
+        if p_checks.archive and \
+          p_checks.requirements and \
```
Comment (richard-jones) [inline server/service/jct/api.coffee:842]: This `if` statement doesn't take into account `p_checks.planS`, and I think it needs to.

The check should pass if:

1. `p_checks.requirements` is false
2. `p_checks.requirements` is true AND `p_checks.planS` is true

File: server/service/jct/api.coffee [hunk 2]
```diff
@@ -736,38 +736,141 @@ API.service.jct.permission = (issn, institution) ->
...
+        else
+          oa_check.OABNonCompliant.push(p_values)
+
+      # return best result
+      pb = undefined
+      if oa_check.OABCompliant.length
+        pb = _best_permission(oa_check.OABCompliant)
+        res.compliant = pb.compliant
+        res.log.push code: 'SA.OABCompliant', parameters: licence: pb.licences, embargo: pb.embargo, version: pb.versions
+        # ToDo - if planS in pb, do we need to add a qualification?
```
Comment (richard-jones) [inline server/service/jct/api.coffee:862]: Not at this stage, no, though it's a question we could post to the team

**Review by richard-jones — CHANGES_REQUESTED** [review]
I think that I have interpreted the code correctly, and there's one area where I think the logic is wrong.  It would help if there was a set of unit test cases for this for the various sets of requirements that might be present on each permission, is that something that we can add?

File: server/service/jct/api.coffee (hunk 1 above)
Comment (anusharanganathan) [reply server/service/jct/api.coffee:838]: No, that logic would be wrong. The plans check was superfluous anyway. I added that incase you wanted a qualification added. I then added that to the values extracted, as that was a better place. So I've just removed checks.plans so as to not confuse you

**Review by richard-jones — APPROVED** [review]
Ok, this looks good to me, though I would love to see some sort of test for it, even if we can just define a couple of real-world test cases to run through it

**Pull request merged**